### PR TITLE
Remove some unused feature flags

### PR DESCRIPTION
**Pull request by mrnugget**
This removes _some_ of the feature flags mentioned in https://github.com/sourcegraph/sourcegraph/issues/45048.

Namely:

- `experimentalFeatures.enablePostSignupFlow`
- `open-beta-enabled` (was already removed)
- `org-code` (was already removed)
- `org-deletion` (removes flag and dotcom-feature-flagged behaviour)
- `github-app-cloud` (that one was unused, so now it's removed)
- `props.authenticatedUser?.tags?.includes('CreateOrg')` (removed)
- `externalServiceUserMode` (removed the whole `cloud-ga.ts` file, since it was unused)
- `AllowUserExternalServicePrivate` user tag (removed, because unused)
- `AllowUserExternalServicePublic` user tag (removed, because unused)

What this PR does **not** remove yet:

- `org-email-invites`

I haven't figured out the code completely yet. I think if the feature flag is off we allow invites only by username. Only if flag is on do we allow invite by email. But I'm not 100% sure yet.

## Test plan

- CI

### CODE REVIEW

**Comment by mrnugget** [comment]
also, lol, not going to merge this before branch cut

**Review by pjlast — APPROVED** [review]

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.00% (-0.03 kb) | -0.03% (-3.65 kb) | -0.03% (-3.63 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/mrn/remove-feature-flags/compare-report.html;tab=live_object) for a full comparison between the commits ccc223d and c7c2a37 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: internal/database/users.go [hunk 1]
```diff
@@ -1468,15 +1468,6 @@ func validPassword(hash, password string) bool {
 	return bcrypt.CompareHashAndPassword([]byte(hash), []byte(password)) == nil
 }
 
-const (
```
Comment (eseliger) [inline internal/database/users.go:1471]: Can we remove tags altogether? This should be done with feature flags these days.

Comment (mrnugget) [reply internal/database/users.go:1471]: I'd love to, yeah

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -95,13 +95,11 @@ type Mutation {
     """
     updateOrganization(id: ID!, displayName: String): Org!
     """
-    Soft or hard deletes an organization.
```
Comment (sashaostrikov) [inline cmd/frontend/graphqlbackend/schema.graphql:98]: Let's add a changelog entry on this API change?

**Review by sashaostrikov — APPROVED** [review]
LGTM!
Added a comment about a changelog entry.

**Review by indradhanush — APPROVED** [review]
Backend changes LGTM.

Frontend changes also look good from my noob eyes.

**Pull request merged**